gammapy/estimators/excess_map.py

```diff
@@ -166,11 +166,14 @@ def run(self, dataset):
 
         axis = MapAxis.from_energy_edges(energy_edges)
 
-        resampled_dataset = dataset.resample_energy_axis(energy_axis=axis)
-
-        # Beware we rely here on the correct npred background in MapDataset.resample_energy_axis
-        resampled_dataset.models = dataset.models
-
+        resampled_dataset = dataset.resample_energy_axis(
+            energy_axis=axis, name=dataset.name
+        )
+        if isinstance(dataset, MapDatasetOnOff):
+            resampled_dataset.models = dataset.models
+        else:
+            resampled_dataset.background = dataset.npred().resample_axis(axis=axis)
```

Comment:
Why not resampling only the `npred_background` and keep the (non-background ) `models` on the resampled dataset? 
I guess it is fully equivalent anyway.

Reply:
It is faster this way because the models don't have to be re-evaluated